### DISCUSSION TITLE
How long does it normally take for someone to respond to "Investigate Issue"?

### DISCUSSION THREAD

**Discussion by nmpu** [zwave-js/zwave-js#6615] [Q&A] [answered]
I reported a [broken configuration](https://github.com/zwave-js/node-zwave-js/discussions/6537) on December 2. It's hard to believe this device configuration was ever tested. It appears to be listening for motion on the wrong endpoint. I know how to create a custom config file, but this problem occurs at a lower level. I'm willing to pursue the fix, but I need to know where to look.

**Comment by kpine** [comment]
Posting again https://github.com/zwave-js/node-zwave-js/issues/6521

**Comment by kpine** [comment]
> It's hard to believe this device configuration was ever tested.

Keep in mind that there are thousands of Z-Wave devices on the market, and there are a couple thousand in the Z-Wave JS database. 
 In addition, some devices are only sold in certain regions, e.g. your Jasco devices are only in the US, while the maintainer of this project is in the EU, so obtaining and testing US devices is harder (feel free to donate devices!). So yes, some devices and their device files are likely to be completely unvalidated in the real world. Often they are often just [imported](https://github.com/zwave-js/node-zwave-js/commits/master/packages/config/config/devices/0x0063/24770_zw4006.json) from other sources (OZW, OpenHAB, ZWA DB).

So realistically, you cannot expect the maintainer of this project to test all Z-Wave devices in all regions. Luckily Z-Wave certification makes a lot of that unnecessary, but there are some edge cases to workaround, such as endpoints.

**Reply by nmpu** [reply]
OK, sounds like a [perfect storm](https://www.merriam-webster.com/dictionary/perfect%20storm). I'm the first person in the world to try to use a **Jasco 24770/ZW4006** with **Z-Wave JS** and I submitted my inquiry 3 days into a 4-month leave. In this case, donating a device would likely make no difference. I need to figure out where the Z-Wave JS source is hiding on my HAOS VM. There's some other things I'd like to tweak. At least it doesn't need to be compiled.

**Reply by AlCalzone** [reply]
You might want to use Z-Wave JS UI for that, which let's you access the config files and test custom ones.

**Reply by nmpu** [reply]
Thanks for stopping by! I've reviewed the [documentation](https://zwave-js.github.io/node-zwave-js/#/config-files/file-format) and already have some custom config files. However, I don't see a way to change the endpoint used for motion detection. That's got to be buried somewhere lower?

**Reply by AlCalzone** [reply]
You'll likely need to use the `preserveEndpoints` or `preserveRootApplicationCCValueIDs` compat flags, maybe `mapRootReportsToEndpoint`. Which one it is really depends on the behavior of the device and I'll have to look at the logs to verify.
https://zwave-js.github.io/node-zwave-js/#/config-files/file-format?id=compat

**Reply by nmpu** [reply]
**Z-Wave JS UI** shows the motion updates coming in for endpoint 1. My other **Jasco** motion devices use endpoint 2. I don't think this is a case of  _reports to the root device are silently ignored_. I don't think it's a mapping problem at all. It's a listening problem. Once the first motion update arrives, a second `_motion_detection` entity appears. I can even create an automation trigger. However, that new entity will not exist on restart and the associated automation will be disabled. I need more than the existing compat flags to solve this problem. It's probably yet another [poorly-behaved Jasco product that passed "certification"](https://community.home-assistant.io/t/beware-of-jasco-z-wave-firmware-updates-which-break-central-scene/648964).

**Reply by AlCalzone** [reply]
Do you have a driver log capturing an interview plus this report?

**Comment by nmpu** [comment]
The [interview](https://github.com/zwave-js/node-zwave-js/files/13537851/interview_node_74.log) and [.json](https://github.com/zwave-js/node-zwave-js/files/13537850/node_74.json) are in my [original inquiry](https://github.com/zwave-js/node-zwave-js/discussions/6537). I just added sample reports for [24770](https://github.com/zwave-js/node-zwave-js/files/14002698/24770_motion_report.log) (broken) and [26931](https://github.com/zwave-js/node-zwave-js/files/14002699/26931_motion_report.log) (working).

For reference, the date code on the **24770** is 1641. These appear to be **Jasco**'s very first Z-Wave [motion switch](https://products.z-wavealliance.org/products/2126). 

Thanks again.

**Comment by AlCalzone** [comment, marked as answer]
Continuing in the original discussion
